docs/building.md

```diff
@@ -31,6 +31,41 @@ make all
 to generate the contract as well as all the example DAOs' initial storage
 defined below.
 
+## Generating a contract metadata
+
+BaseDAO metadata should be compliant with [TZIP-16](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md).
+
+
+To print the metadata you'll need the `stack` tool, see
+[The Haskell Tool Stack](https://docs.haskellstack.org/en/stable/README/) tutorial
+for instructions on how to obtain it.
+
+```bash
+make metadata > metadata.json
+```
+
+This metadata should be stored either:
+- On IPFS: [Get started with IPFS](https://ipfs.io/)
+- As a separate contract: [The `tezos-storage` URI Scheme](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#the-tezos-storage-uri-scheme)
+
+
+To use this metadata, follow [TZIP-16 Contract storage](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-16/tzip-16.md#contract-storage)
+
+Specifically:
+> The encoding of the values must be the direct stream
+of bytes of the data being stored. For instance, an URI starting with `http:`
+will start with the 5 bytes `0x687474703a` (`h` is `0x68`, `t` is `0x74`,
+etc.). There is no implicit conversion to Michelson's binary format (`PACK`) nor
+quoting mechanism.
+
+After the conversion, the bytes could be use in the argument `metadata_map` of the `make <storage>` command
+(See [Storing](#storing) section for more detail.)
```

Comment:
Oh I see now that you mention the storing section here, however with the reordering you can point at the `Generating a contract storage` section or even just say "the next section", given that it immediately follows.